src/main/java/gov/cabinetoffice/gap/adminbackend/services/SchemeEditorService.java

```diff
@@ -81,5 +87,30 @@ private List<String> getEmailsFromUserSubBatch(final List<String> userSubs, fina
                 awsEncryptionService.decryptField(item.emailAddress())).toList();
     }
 
+    public SchemeEntity addEditorToScheme(Integer schemeId, String editorEmailAddress, String jwt) {
+        AdminSession session = HelperUtils.getAdminSessionForAuthenticatedUser();
+        SchemeEntity scheme = this.schemeRepo.findById(schemeId).orElseThrow(EntityNotFoundException::new);
+        List<GrantAdmin> existingEditors = scheme.getGrantAdmins();
+        GrantAdmin editorToAdd;
+        try {
+            editorToAdd = userService.getGrantAdminIdFromUserServiceEmail(editorEmailAddress, jwt);
+        } catch (Exception e) {
+            throw new FieldViolationException("editorEmailAddress", "This account does not have an 'Administrator' account.");
+        }
+
+        if (existingEditors.stream().anyMatch(editor -> editor.getId().equals(editorToAdd.getId()))) {
```

Comment:
do we need to throw an exception here or could we just return without performing the save? Would make the method more idempotent

Reply:
Throwing the FieldViolationException is required for the front end to display the in-line errors in the editorEmailAddress box

Reply:
Is there a requirement for that error message in the ACs? Probably not worth me dying on a hill over but I think making the entire journey idempotent makes more sense. If this is whats been asked for then just resolve the comment tho.